### PR TITLE
Fix link sections and include images/pictures

### PR DESCRIPTION
**Pull request by raconte**
All the documents has been modified in order to link properly the corresponding sections and to show the pictures of each section

### CODE REVIEW

**Comment by fgalan** [comment]
LGTM (but don't merge yet... I'm going to piggyback some additional changes in this PR related with documetnation for releaseing 0.23.0)

File: README.md [hunk 1]
```diff
@@ -18,21 +18,23 @@ Orion is a C++ implementation of the NGSI9/10 REST API binding developed as a pa
 
 Orion Context Broker allows you to manage all the whole lifecycle of context information including updates, queries, registrations and subscriptions. It is an NGSI9/10 server implementation to manage context information and its availability. Using the Orion Context Broker, you are able to register context elements and manage them through updates and queries. In addition, you can subscribe to context information so when some condition occurs (e.g. an interval of time has passed or the context elements have changed) you receive a notification. These usage scenarios and the Orion Context Broker features are described in this document.
 
-If this is your first approach with Orion Context Broker, it is highly recommended to have a look to the brief [Quick Start guide](doc/md/quick_start_guide.md).
+If this is your first approach with Orion Context Broker, it is highly recommended to have a look to the brief [Quick Start guide](doc/manuals/quick_start_guide.md).
```
Comment (kzangeli) [inline README.md]: I think I commented on this some time ago ...

I believe "contact" is a better choice that "approach" (and a 'the'):
  If this is your first contact with the Orion Context Broker...

File: README.md [hunk 2]
```diff
@@ -18,21 +18,23 @@ Orion is a C++ implementation of the NGSI9/10 REST API binding developed as a pa
...
 
 ### Requirements
 
-* System resources: see [these recommendations](doc/md/admin/resources.md#resources-recommendations)
+* System resources: see [these recommendations](doc/manuals/admin/resources.md#resources-recommendations)
 * Operating system: CentOS/RedHat. The reference operating system is CentOS 6.3  
 but it should work also in any later CentOS/RedHat 6.x version.
-* Database: MongoDB is required running either in the same system where Orion Context Broker is going to be installed or in a different host accessible through the network. The recommended MongoDB version is 2.6.9 (although it should work with later MongoDB 2.6.x and 3.0.x versions). It is not recommended using MongoDB 2.4.x., as some [geolocated queries](doc/md/user/geolocation.md).
+* Database: MongoDB is required running either in the same system where Orion Context Broker is going to be installed or in a different host accessible through the network. The recommended MongoDB version is 2.6.9 (although it should work with later MongoDB 2.6.x and 3.0.x versions). It is not recommended using MongoDB 2.4.x., as some [geolocated queries](doc/manuals/user/geolocation.md) may not work.
```
Comment (kzangeli) [inline README.md]: English:
 * is required to run
 * It is not recommended to use MongoDB 2.4.x ...

Actually, I would like to rephrase the first sentence like this:
*Database: MongoDB is required to run either in the same host where Orion Context Broker is to be installed or in a different host accessible through the network.*

File: README.md [hunk 3]
```diff
@@ -247,17 +249,17 @@ In order to update one of the entity atributes (temperature):
     }
     EOF
 
-Please have a look to the [Quick Start guide](doc/md/quick_start_guide.md) if you want to test these operations in an actual public instance of Oriion Context Broker. In addition, have a look to the API Walkthrough and API Reference sections below in order to know more details about the API (subscriptions, registrations, etc.).
+Please have a look to the [Quick Start guide](doc/manuals/quick_start_guide.md) if you want to test these operations in an actual public instance of Oriion Context Broker. In addition, have a look to the API Walkthrough and API Reference sections below in order to know more details about the API (subscriptions, registrations, etc.).
```
Comment (kzangeli) [inline README.md]: * Please have a look *at*
* Oriion ...

File: doc/manuals/admin/database_admin.md [hunk 4]
```diff
@@ -68,8 +68,7 @@ to take into account:
 
 Normally, Orion Context Broker uses just one database at MongoDB level
 (the one specified with the `-db` command line option, typically "orion").
-However, when [multitenant/multiservice is
-used](Publish/Subscribe_Broker_-_Orion_Context_Broker_-_User_and_Programmers_Guide#Multi_service_tenancy "wikilink")
+However, when [multitenant/multiservice](#multitenant/multiservice-database-separation) is used
```
Comment (kzangeli) [inline doc/manuals/admin/database_admin.md]: comma after 'used'

File: doc/manuals/admin/database_admin.md [hunk 5]
```diff
@@ -132,9 +131,7 @@ indexes could be recommended:
 
 The only index that Orion Context Broker actually ensures is the
 "2dsphere" one in the location.coords field in the entities collection,
-due to functional needs (in order [geo-location
-functionality](Publish/Subscribe_Broker_-_Orion_Context_Broker_-_User_and_Programmers_Guide#Geolocation_capabilities "wikilink")
-to work). The index is ensured at Orion startup or when entities are
+due to functional needs [geo-location functionality](../user/geolocation.md)to work. The index is ensured at Orion startup or when entities are
 created for first time.
```
Comment (kzangeli) [inline doc/manuals/admin/database_admin.md]: for **the** first time

File: doc/manuals/admin/database_admin.md (hunk 4 above)
Comment (LeandroGuillen) [inline doc/manuals/admin/database_admin.md]: I disagree.

File: doc/manuals/admin/database_admin.md [hunk 6]
```diff
@@ -151,7 +148,7 @@ particular set up, but the results in your particular environment may
 differ depending on hardware profile, the particular entities being used
 for the test, set up situation, etc. In this particular case, the
 resources of the system under test (a VMware-based VM) are: 2 vCPU (on a
-physical host based on Intel Xeon E5620@2.40GHz) and 4GB RAM. Both Orion
+physical host based on Intel Xeon E5620@ 2.40GHz) and 4GB RAM. Both Orion
```
Comment (kzangeli) [inline doc/manuals/admin/database_admin.md]: I wouldn't add that space after '@' ...

File: doc/manuals/admin/diagnosis.md [hunk 7]
```diff
@@ -130,9 +130,9 @@ The symptoms of a database connection problem are the following ones:
 
 In both cases, check that the connection to MonogDB is correctly
 configured (in particular, the BROKER\_DATABASE\_HOST if you are running
-Orion Context Broker [as a service](#As_system_service "wikilink") or
+Orion Context Broker [as a service](../../../README.md#as-system-service) or
 the "-dbhost" option if you are running it [from the command
-line](#From_the_command_line "wikilink")) and that the mongod or mongos
+line](cli.md)) and that the mongod or mongos
```
Comment (kzangeli) [inline doc/manuals/admin/diagnosis.md]: mongod/mongos  instead of "mongod or mongos"

**Comment by kzangeli** [comment]
Picture doc/manuals/admin/Orion-ioflows.png:  the process to forward to doesn't *have to be* an Orion, right?

File: doc/manuals/admin/database_admin.md (hunk 4 above)
Comment (kzangeli) [inline doc/manuals/admin/database_admin.md]: It is a pretty common mistake (in my opinion it is definitely a mistake) that often makes the sentence next to impossible to understand ...

**Comment by kzangeli** [comment]
Picture doc/manuals/user/Federation.png:
Contex -> Context

**Comment by kzangeli** [comment]
Picture doc/manuals/user/ServicePathWithMultiservice.png:

The "(-multiservice)" I imagine was added to clarify that the option was needed to have the broker in multi-tenancy mode.  Now the broker starts in multi-tenancy mode by default.

Perhaps the "clarification" should be removed?

File: doc/manuals/admin/database_admin.md (hunk 4 above)
Comment (LeandroGuillen) [inline doc/manuals/admin/database_admin.md]: Commas are mandatory only if the sentence is trying to help understand the main sentence, which is "the behavior is different and... ", and is completely optional. See that without the first sentence the second would not make sense, hence the absence of a comma.

File: doc/manuals/user/federation.md [hunk 8]
```diff
@@ -148,11 +148,10 @@ attribute values) in C. You can check it by doing a queryContext to C:
 
 In the current context broker version, the semantics of
 nofityContextRequest are the same that [updateContext
-APPEND](#Entity_Creation "wikilink") or, if the context element already
-exist, the semantics of [updateContext
-UPDATE](#Update_context_elements "wikilink"). Thus, federation doesn't
-provide exact mirroring: a updateContext DELETE to one context broker
-will not produce the same effect in the federated context broker.
+APPEND  or, if the context element already
+exist, the semantics of updateContext UPDATE](walkthrough_apiv1.md#update-context-elements).
```
Comment (kzangeli) [inline doc/manuals/user/federation.md]: exist -> exists

File: doc/manuals/user/federation.md [hunk 9]
```diff
@@ -148,11 +148,10 @@ attribute values) in C. You can check it by doing a queryContext to C:
...
+Thus, federation doesn't provide exact mirroring: a updateContext DELETE to
```
Comment (kzangeli) [inline doc/manuals/user/federation.md]: a -> an

File: doc/manuals/user/security.md [hunk 10]
```diff
@@ -2,7 +2,7 @@
 
 ## Authentication and Authorization 
 
-Orion doesn't provide "native" authentication or authorization mechanisms to enforce access control. However, they can be achieved [the access control framework provided by FIWARE GEs](FIWARE.ArchitectureDescription.Security.Access_Control_Generic_Enabler "wikilink").
+Orion doesn't provide "native" authentication or authorization mechanisms to enforce access control. However, they can be achieved [the access control framework provided by FIWARE GEs](https://forge.fiware.org/plugins/mediawiki/wiki/fiware/index.php/FIWARE.ArchitectureDescription.Security.Access_Control_Generic_Enabler).
```
Comment (kzangeli) [inline doc/manuals/user/security.md]: * or -> nor any
* they -> authentication/authorization

File: doc/manuals/admin/database_admin.md (hunk 4 above)
Comment (fgalan) [inline doc/manuals/admin/database_admin.md]: No clear agreement on this, so I'd leave as it is now :)

NTC

File: README.md (hunk 1 above)
Comment (fgalan) [inline README.md]: Yes, you have commented the same in other documents. But I forgot ;)

Fixed in fdf0ebc

File: README.md (hunk 2 above)
Comment (fgalan) [inline README.md]: Fixed in fdf0ebc

File: README.md (hunk 3 above)
Comment (fgalan) [inline README.md]: Fixed in fdf0ebc

File: doc/manuals/admin/database_admin.md (hunk 5 above)
Comment (fgalan) [inline doc/manuals/admin/database_admin.md]: Fixed in fdf0ebc

File: doc/manuals/admin/database_admin.md (hunk 6 above)
Comment (fgalan) [inline doc/manuals/admin/database_admin.md]: Fixed in fdf0ebc

File: doc/manuals/admin/diagnosis.md (hunk 7 above)
Comment (fgalan) [inline doc/manuals/admin/diagnosis.md]: Fixed in fdf0ebc

File: doc/manuals/user/federation.md (hunk 8 above)
Comment (fgalan) [inline doc/manuals/user/federation.md]: Fixed in fdf0ebc

File: doc/manuals/user/federation.md (hunk 9 above)
Comment (fgalan) [inline doc/manuals/user/federation.md]: Fixed in fdf0ebc

File: doc/manuals/user/security.md (hunk 10 above)
Comment (fgalan) [inline doc/manuals/user/security.md]: Fixed in fdf0ebc

**Comment by fgalan** [comment]
Regarding comments about images, let's leave that out of scope of this PR by the moment (images are not as easy to change as text).

**Comment by kzangeli** [comment]
LGTM

**Pull request merged**